js/multimodaltextbox/shared/MultimodalTextbox.svelte

```diff
@@ -0,0 +1,484 @@
+		tick
+	} from "svelte";
+	import { Upload } from "@gradio/upload";
+	import { Image } from "@gradio/image/shared";
+	import type { FileData } from "@gradio/client";
+	import { Copy, Check, Clear, File, Music, Video } from "@gradio/icons";
+	import { fade } from "svelte/transition";
+	import type { SelectData } from "@gradio/utils";
+
+	export let value: { type: string; text?: string; file?: FileData | null }[] =
+		[];
+	export let value_is_output = false;
+	export let lines = 1;
+	export let placeholder = "Type here...";
+	export let disabled = false;
+	export let show_label = true;
+	export let container = true;
+	export let max_lines: number;
+	export let show_copy_button = false;
+	export let rtl = false;
+	export let autofocus = false;
+	export let text_align: "left" | "right" | undefined = undefined;
+	export let autoscroll = true;
+	export let root: string;
+	export let file_types: string[] | null = null;
+
+	let el: HTMLTextAreaElement | HTMLInputElement;
+	let copied = false;
+	let timer: NodeJS.Timeout;
+	let can_scroll: boolean;
+	let previous_scroll_top = 0;
+	let user_has_scrolled_up = false;
+
+	let text = "";
+	let dragging = false;
+	$: dispatch("drag", dragging);
+
+	let accept_file_types: string | null;
+	if (file_types == null) {
+		accept_file_types = null;
+	} else {
+		file_types = file_types.map((x) => {
+			if (x.startsWith(".")) {
+				return x;
+			}
+			return x + "/*";
+		});
+		accept_file_types = file_types.join(", ");
+	}
+
+	$: value, el && lines !== max_lines && resize({ target: el });
+	$: file_count = Array.isArray(value) ? value.filter((item) => item.type === "file").length : 0;
+
+	const dispatch = createEventDispatcher<{
+		change: typeof value;
+		submit: typeof value;
+		blur: undefined;
+		select: SelectData;
+		input: undefined;
+		focus: undefined;
+		drag: boolean;
+		upload: FileData[] | FileData;
+		clear: undefined;
+		load: FileData[] | FileData;
+		error: string;
+	}>();
+
+	beforeUpdate(() => {
+		can_scroll = el && el.offsetHeight + el.scrollTop > el.scrollHeight - 100;
+	});
+
+	const scroll = (): void => {
+		if (can_scroll && autoscroll && !user_has_scrolled_up) {
+			el.scrollTo(0, el.scrollHeight);
+		}
+	};
+
+	function handle_change(): void {
+		dispatch("change", value);
+		if (!value_is_output) {
+			dispatch("input");
+		}
+	}
+
+	afterUpdate(() => {
+		if (autofocus && el !== null) {
+			el.focus();
+		}
+		if (can_scroll && autoscroll) {
+			scroll();
+		}
+		value_is_output = false;
+	});
+	$: value, handle_change();
+
+	async function handle_copy(): Promise<void> {
+		if ("clipboard" in navigator) {
+			const text = value.find((item) => item && "text" in item);
+			if (text && "text" in text && typeof text["text"] === "string") {
+				await navigator.clipboard.writeText(text["text"]);
+				copy_feedback();
+			}
+		}
+	}
+
+	function copy_feedback(): void {
+		copied = true;
+		if (timer) clearTimeout(timer);
+		timer = setTimeout(() => {
+			copied = false;
+		}, 1000);
+	}
```

Comment:
We should move this as well but it will need refactoring into a store:

```js
import { writable } from 'svelte/store';

export function make_copy_store() {
  const copied = writeable(false);
  
  async function copy(value): Promise<void> {
    if ("clipboard" in navigator) {
      const text = value.find((item) => item && "text" in item);
      if (text && "text" in text && typeof text["text"] === "string") {
        await navigator.clipboard.writeText(text["text"]);
          copy_feedback();
      }
    }
  }
  
  function copy_feedback(): void { 
    copied.set(true);
    if (timer) clearTimeout(timer);
      timer = setTimeout(() => {
        copied.set(false);
      }, 1000);
  }
  
  return {
    copied,
    copy
  }
}
```

Then we should just able to use it like:

```
import { make_copy_store } from './wherever';

const { copied, copy } = make_copy_store();

// to copy
copy(value)

// the live 'copied' state, we'll need to replace all instances of `copied` with `$copied`.
$copied